frontend/src/Language/Granule/Syntax/Expr.hs

```diff
@@ -94,6 +94,7 @@ data ExprF ev a expr value =
      -- let p <- e1 in e2
   | ValF Span a value
   | CaseF Span a expr [(Pattern a, expr)]
+  | HoleF Span a
```

Comment:
Wouldn't it be nicer to add holes via `ExtF` to ensure we don't have any holes left after type checking?